sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs

```diff
@@ -2887,6 +2887,224 @@ public new PageBlobClient WithVersion(string versionId)
             }
         }
         #endregion UploadPagesFromUri
+
+        #region OpenWrite
+        /// <summary>
+        /// Opens a stream for writing to the blob.
+        /// </summary>
+        /// <param name="overwrite">
+        /// Whether an existing blob should be deleted and recreated.
+        /// </param>
+        /// <param name="position">
+        /// The offset within the blob to begin writing from.
+        /// </param>
+        /// <param name="size">
+        /// Required if overwrite is set to true, or the underlying
+        /// Page Blob is being created for the first time.
+        /// Specifies the size of the new Page Blob.
+        /// </param>
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A stream to write to the Append Blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+#pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual Stream OpenWrite(
+#pragma warning restore AZC0015 // Unexpected client method return type.
+            bool overwrite,
+            long position,
+            long? size = default,
```

Comment:
I chose to add an optional size parameter, and mentioned in the xml comments it is required if overwrite is set to true, or the Page Blob is being created for the first time.

Reply:
Moved `Size` to `PageBlobOpenWriteOptions`.  I'd like to keep `position` a required parameter to prevent users from overwriting the data at the beginning of a Page Blob by default`.